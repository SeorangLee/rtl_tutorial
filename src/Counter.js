import React from "react";
import { useSelector, useDispatch } from "react-redux";
import {Link} from 'react-router-dom';

function Counter(){

    const dispatch = useDispatch()

	const { value } = useSelector(state => state.value)
	const { count } = useSelector(state => state.count)

	const addValue = () => {
		dispatch({ type: 'increment' })
	}
	const subValue = () => {
		dispatch({ type: 'decrement' })
	}
	const resetValue = () => {
		dispatch({ type: 'reset' })
	}
	const pushButton = () => {
		dispatch({ type: 'push' })
	}

    return(
        <div className="App">
			<div>
				value: {value}
			</div>
			<button onClick={addValue}> + </button>
			<button onClick={subValue}> - </button>
			<button onClick={resetValue}> reset </button>
			<div>
				count: {count}
			</div>
			<button onClick={pushButton}> click </button>
            <Link to="/">Go to Login</Link>
            
		</div>
    )
}

export default Counter;