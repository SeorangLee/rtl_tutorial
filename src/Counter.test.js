import React from "react";
import { screen, render } from "@testing-library/react";
import { createStore } from "redux";
import reducer from "./reducer";
import store from "./store";
import Counter from "./Counter";
import {Provider} from 'react-redux'



const Wrapper = ({children}) => (
    <Provider store={store}>{children}</Provider>
)

describe("<Counter />", ()=>{
    it("Counter snap shot", ()=>{
        const utils = render(<Counter />, {wrapper:Wrapper});
        expect(utils.container).toMatchSnapshot();
    });
})