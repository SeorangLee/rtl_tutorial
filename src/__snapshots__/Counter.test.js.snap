// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> Counter snap shot 1`] = `
<div>
  <div
    class="App"
  >
    <div>
      value: 
      0
    </div>
    <button>
       + 
    </button>
    <button>
       - 
    </button>
    <button>
       reset 
    </button>
    <div>
      count: 
      0
    </div>
    <button>
       click 
    </button>
  </div>
</div>
`;
